spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -53,11 +55,17 @@
 
 	private final ChannelProvider channelProvider;
 
+	private final CredentialsProvider credentialsProvider;
+
 	public DefaultPublisherFactory(String projectId, ExecutorProvider executorProvider,
-			ChannelProvider channelProvider) {
+			CredentialsProvider credentialsProvider) {
 		this.projectId = projectId;
 		this.executorProvider = executorProvider;
-		this.channelProvider = channelProvider;
+		this.channelProvider = TopicAdminSettings.defaultChannelProviderBuilder()
```

Comment:
OK. One more time: we develop here Open Source and that wouldn't be so friendly to hide from end-user the possibility to inject any existing public `ChannelProvider` implementation. Even if that would be only for testing (mock), that is already end-user choice.

As far as I would insist to stop to chase that `clientLib` header. It is still open source and that is already end-user choice to show or hide such an option from the target platform.

Anyway he can't do that so clearly because he/she will always authenticate to the platform and we there always can track and trace what and how he/she does with our platform. As I suggested before, something like `SpringChannelProvider` wrapper may server the `clientLib` wishes, but can't be as a strong requirement.

If you really want to force that option, the GCP should reconsider it to something like a special file in the classpath. Or as I suggested once, too - via SSL key what end-users get from the platform together with the account.